pkg/vgmanager/devices.go

```diff
@@ -200,11 +200,14 @@ func (r *VGReconciler) getNewDevicesToBeAdded(ctx context.Context, blockDevices
 	return validBlockDevices, nil
 }
 
-func isDeviceAlreadyPartOfVG(vgs []lvm.VolumeGroup, diskName string, volumeGroup *lvmv1alpha1.LVMVolumeGroup) bool {
-	for _, vg := range vgs {
-		if vg.Name == volumeGroup.Name {
-			for _, pv := range vg.PVs {
-				if pv.PvName == diskName {
+func isDeviceAlreadyPartOfVG(nodeStatus *lvmv1alpha1.LVMVolumeGroupNodeStatus, diskName string, volumeGroup *lvmv1alpha1.LVMVolumeGroup) bool {
+	if nodeStatus == nil {
```

Comment:
If node status is nil that means we never checked the vgs right? So is it unsafe to return false here?

Reply:
This is only a safeguard for the tests. `nodeStatus` is created if not found at the beginning of the reconciliation, so it will never be nil.